src/app/classes/stix/stix-object.ts

```diff
@@ -750,6 +697,33 @@ export abstract class StixObject extends Serializable {
     return result;
   }
 
+  /**
+   * Checks if the provided field has a valid value.
+   * Returns false for undefined, null, empty string, or empty array.
+   * @param {*} field - The value to validate.
+   * @returns {boolean} - True if the field has a value, false otherwise.
+   */
+  public hasValue(field) {
+    return (
+      field !== undefined &&
+      field !== null &&
+      field !== '' &&
+      !(Array.isArray(field) && field.length === 0)
+    );
+  }
+
+  /**
+   * Filters the properties of an object, returning a new object containing only
+   * those entries whose values pass the hasValue check.
+   * @param {Object} obj - The object to filter.
+   * @returns {Object} - A new object with only the valid entries.
+   */
+  public filterObject(obj) {
+    return Object.fromEntries(
+      Object.entries(obj).filter(entry => this.hasValue(entry[1]))
+    );
+  }
```

Comment:
Suggest renaming this function to something more descriptive like `removeEmptyFields`.

This function will filter out top-level values correctly, but not nested objects or arrays. For example, the Asset SDO's `x_mitre_related_assets` array has optional `related_asset_sectors: string[]` and `description: string` fields, which will not be removed if no values are present:

```
        "x_mitre_related_assets": [
            {
                "name": "Intermediate System",
                "related_asset_sectors": [],
                "description": ""
            }
        ],
```